src/main.ts

```diff
@@ -48,34 +73,79 @@ async function run(): Promise<void> {
       );
     }
 
-    const ignores = ignore();
+    // Compute the absolute root and compute the glob.
+    const [absoluteRoot, computedGlob] = await absoluteRootAndComputedGlob(root, glob);
+
+    // Build complete file list.
+    const files = await expandGlob(absoluteRoot, computedGlob);
+
+    // Process ignores:
+    //
+    // - Find ignore file
+    // - Format all files to be posix relative to input.path
+    // - Filter out items that match
     if (processGcloudIgnore) {
-      const ignoreList = await parseGcloudIgnore('.gcloudignore');
-      ignores.add(ignoreList);
+      const ignores = ignore();
+
+      // Look for a .gcloudignore in the repository root.
+      if (process.env.GITHUB_WORKSPACE) {
+        const gcloudIgnorePath = path.join(process.env.GITHUB_WORKSPACE, '.gcloudignore');
+        const ignoreList = await parseGcloudIgnore(gcloudIgnorePath);
+
+        if (ignoreList.length) {
+          core.debug(`Used .gcloudignore at: ${gcloudIgnorePath}`);
+          core.debug(`parsed ignore list: ${JSON.stringify(ignoreList)}`);
+
+          ignores.add(ignoreList);
+        }
+      }
+
+      for (let i = 0; i < files.length; i++) {
```

Comment:
Do we need this if `ignoreList` is empty as `Ignore.ignores()` would always be false?

Reply:
We could pull the above into a `processGcloudIgnore([]string):[]string` fn that we can test more easily.

Reply:
Good point about only iterating over files if we actually have an ignore to process. Just pushed that up. I'll refactor in the next commit.

Reply:
Actually, I'm going to leave it as-is. It's already pretty easy to test, and moving it to a function gets unnecessarily messy.